txscript/engine.go

```diff
@@ -718,7 +718,7 @@ func NewEngine(scriptPubKey []byte, tx *wire.MsgTx, txIdx int, flags ScriptFlags
 
 	// The signature script must only contain data pushes for P2SH which is
 	// determined based on the form of the public key script.
-	if isAnyKindOfScriptHash(scriptPubKey) {
+	if isAnyKindOfScriptHash(scriptPubKey, vm.hasFlag(ScriptVerifyTreasury)) {
```

Comment:
Hmm.  I guess this is fine, but I think it would be nicer to just make the function a method on `Engine` and do the `if vm.hasFlag(ScriptVerifyTreasury)` inside there so the function signature doesn't have to change (aside from adding `vm.` in front of the callers).